### PR TITLE
NIFI-9058 Core attributes shall not be filtered from Attributes List

### PR DESCRIPTION
**Pull request by lordgamez**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

Currently "Include Core Attributes" property filters out any core attributes specified in the "Attributes List" property, but does not filter any attributes matching the regular expression in the "Attributes Regular Expression" property. This makes the functionality of the processor inconsistent. "Include Core Attributes" should not filter out specifically listed attributes in the "Attributes List" property, and this PR removes this filtering.

https://issues.apache.org/jira/browse/NIFI-9058

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [X] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [X] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [X] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [X] Have you verified that the full build is successful on JDK 11?
- [X] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [X] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [X] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [X] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [X] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AttributesToJSON.java [hunk 1]
```diff
@@ -212,10 +212,7 @@ protected void init(final ProcessorInitializationContext context) {
             if (ats != null) {
                 Set<String> result = new HashSet<>(ats.length);
                 for (String str : ats) {
-                    String trim = str.trim();
-                    if (!atrsToExclude.contains(trim)) {
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AttributesToJSON.java:216]: With this change, the `atrsToExclude` parameter no longer appears to be used, so the method signature should be adjusted and calling code adjusted.

**Review by exceptionfactory — COMMENTED** [review]
Thanks for keeping this PR going @lordgamez! It looks straightforward in general, but I noted one question regarding the parameters for `buildAtrs` that no longer appears necessary.

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AttributesToJSON.java (hunk 1 above)
Comment (lordgamez) [reply nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AttributesToJSON.java:216]: Thanks for the review! I updated the signature in ff81c1c810bb972ebe7de9d9915fc024a76ec0f2

**Review by exceptionfactory — APPROVED** [review]
Thanks for addressing the feedback @lordgamez, this looks good. +1 Merging.

**Pull request closed without merging**